File: app/src/main/java/pl/butajlo/androidadvanced/data/TrendingReposResponse.java
package pl.butajlo.androidadvanced.data;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.Json;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

import java.util.List;

import pl.butajlo.androidadvanced.models.Repo;

@AutoValue
public abstract class TrendingReposResponse {

    @Json(name = "items")
    public abstract List<Repo> repos();

    public static JsonAdapter<TrendingReposResponse> jsonAdapter(Moshi moshi) {
        return new AutoValue_TrendingReposResponse.MoshiJsonAdapter(moshi);
    }

}

File: app/src/main/java/pl/butajlo/androidadvanced/data/TrendingReposResponse.kt
package pl.butajlo.androidadvanced.data

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass
import pl.butajlo.androidadvanced.models.Repo

@JsonClass(generateAdapter = true)
data class TrendingReposResponse(
        @Json(name = "items") val repos: List<Repo>
)